### ISSUE TITLE
Consider allowing custom messages when started as an actor.

### ISSUE THREAD

**Issue by hayleigh-dot-dev** [pta2002/gleam-filespy#3]
One of the improvements `mist` made a while back with its web socket processes was the ability to send custom messages to the socket's actor. This meant you could properly do pubsub things and have some other process push events to clients.

I think it would be really great if filespy had similar functionality. Concretely, I'm experimenting using filespy in lustre's dev tools to add hot reloading to its development server. I have things working (pending that PR you merged recently being published) but I have to set up a separate actor that sits between filespy and connected web socket clients. This is because although the filespy actor can have custom state, there's no way to communicate to the actor besides change events. I currently have something like:

```
|---------|              |-------------|                 |--------------|
|         |              |             |<-- connection --|              |
| filespy |--- events -->| coordinator |                 | websocket(s) |
|         |              |             |---- events ---->|              |
|---------|              |-------------|                 |--------------|

```

I could cut out this coordinator actor entirely if I had a way to tell filespy that a new client has connected. In mist they support it like this:

```gleam
pub type WebsocketMessage(custom) {
  Text(String)
  Binary(BitArray)
  Closed
  Shutdown
  Custom(custom)
}
```

I think the easiest change would be to just expand the `Change` type with a second `Custom` variant – it'd make the actor stuff much more useful!

**Comment by pta2002** [comment]
So in effect this'd allow you to send a custom message to the same actor that handles file changes, right? Sorry, as you can probably tell I haven't touched this in a while so I'm a bit rusty on the details.

I have no opposition to this, I made filespy more as a proof of concept to see if I could hack hot code reloading into gleam, and if this is useful for that goal then great :)

PS: Life has gotten in the way (I started a new job + I'm trying to finish my thesis at the same time) and I haven't touched this in a while, and I don't want to have to be blocking your progress with changes I'm going to take forever to get around to, so if you'd like I can transfer ownership of the repo to you (especially since you helped me set this up in the first place :p)

**Comment by hayleigh-dot-dev** [comment]
> I don't want to have to be blocking your progress

No worries at all! We're all doing this as a hobby after all ^.^ Good luck on your thesis 💕

**Comment by pta2002** [comment]
Fixed by #4. I've created a release @hayleigh-dot-dev, hope it's useful!
